lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -67,17 +68,33 @@ def build_index(self, server_config, dates):
             self.index_version,
             self.index_prefix,
         )
-        index = "/"
+        index = ""
         for d in dates:
             index += f"{idx}{d},"
-        return index
+        if not index:
+            drb = Dataset.attach(controller="node", name="drb")
+            index_map = Metadata.getvalue(dataset=drb, key="server.index-map")
+            for key, values in index_map.items():
+                if self.payload["run_id"] in values:
+                    index = f"{key}"
+        return f"/{index}"
+
+    def build_index_from_metadata(self, find_metadata):
+        drb = Dataset.attach(controller="node", name="drb")
+        index_map = Metadata.getvalue(dataset=drb, key="server.index-map")
+        for key, values in index_map.items():
+            if self.payload["run_id"] in values:
+                return "/" + key
+        return None
 
     def date_range(self, start: AnyStr, end: AnyStr) -> list:
         """
         Builds list of range of dates between start and end
         It expects the date to look like YYYY-MM
         """
         date_range = []
+        if not start or not end:
+            return date_range
```

Comment:
If what you want to do here is to return an empty list if either `start` or `end` is blank or omitted, then you should just do that.  Returning the value of `data_range` accomplishes the same thing, but it's slightly obfuscating for the reader, and doing the test first saves Python from having to set up `date-range` if you're going to bail out early.
```
        if not start or not end:
            return []
        date_range = []
```